File: order_service_producer/src/main/java/com/github/marceloedudev/domain/entity/BearerToken.java
package com.github.marceloedudev.domain.entity;

import com.github.marceloedudev.domain.errors.base.NotificationErrors;

import java.util.List;

public class BearerToken {

    private String value;

    private NotificationErrors notifications;

    public BearerToken(String value) {
        this.value = value;
        notifications = new NotificationErrors();
    }

    public String getValue() {
        return value.replace("Bearer ", "");
    }

    public boolean validate() {
        if (value == null || value.isEmpty() || value.isBlank()) {
            notifications.addError("Invalid token");
        }
        return notifications.hasErrors();
    }

    public List<String> getNotifications() {
        return notifications.getErrorsMessage();
    }

}

File: auth_service/src/domain/entity/BearerToken.ts
import NotificationContext from "../errors/base/NotificationErrors";

export default class BearerToken {
    private notifications: NotificationContext;

    private token: string;

    constructor(bearerHeader) {
        let token;
        if (bearerHeader !== undefined) {
            [, token] = this.explodeBearerHeader(bearerHeader);
        }
        this.token = token;
        this.notifications = new NotificationContext();
    }

    public validate(): boolean {
        if (this.token === undefined) {
            this.notifications.addError("Forbidden");
        }
        return this.notifications.hasErrors();
    }

    public getNotifications(): string[] {
        return this.notifications.getErrors();
    }

    public getToken(): string {
        return this.token;
    }

    private explodeBearerHeader(bearerHeader): Array<string | undefined> {
        try {
            return bearerHeader.split(" ");
        } catch (error) {
            return [undefined, undefined];
        }
    }
}
